middleware/fastcgi/fcgiclient.go

```diff
@@ -373,6 +374,14 @@ func (this *FCGIClient) Request(p map[string]string, req io.Reader) (resp *http.
 	}
 	resp.Header = http.Header(mimeHeader)
 
+	if resp.Header.Get("Status") != "" {
```

Comment:
Did you try `resp.Status` and `resp.StatusCode` first? If not, go ahead and see if those have the values you would expect. If that works, we shouldn't need to separate them ourselves.